packages/graph/hash_graph/lib/graph/src/shared/subgraph/edges/mod.rs

```diff
@@ -1,175 +1,126 @@
 use std::collections::{hash_map::Entry, HashMap, HashSet};
 
-use serde::Serialize;
-use type_system::uri::BaseUri;
-use utoipa::{openapi, ToSchema};
-
-use crate::identifier::{
-    knowledge::{EntityId, EntityVersion},
-    ontology::OntologyTypeVersion,
-    GraphElementEditionId,
-};
+use crate::identifier::{knowledge::EntityEditionId, ontology::OntologyTypeEditionId};
 
 mod edge;
 mod kind;
 
 pub use self::{
-    edge::{GenericOutwardEdge, KnowledgeGraphOutwardEdges, OntologyOutwardEdges, OutwardEdge},
+    edge::{KnowledgeGraphOutwardEdges, OntologyOutwardEdges, OutwardEdge},
     kind::{KnowledgeGraphEdgeKind, OntologyEdgeKind, SharedEdgeKind},
 };
 
-#[derive(Default, Debug, Serialize)]
-#[serde(transparent)]
-pub struct OntologyRootedEdges(
-    // TODO: expose it through methods instead of making this field `pub`
-    //   see https://app.asana.com/0/1202805690238892/1203358665695491/f
-    pub HashMap<BaseUri, HashMap<OntologyTypeVersion, HashSet<OntologyOutwardEdges>>>,
-);
-
-// This is needed because Utoipa doesn't know how to handle the BTreeMaps
-impl ToSchema for OntologyRootedEdges {
-    fn schema() -> openapi::Schema {
-        openapi::ObjectBuilder::new()
-            .additional_properties(Some(openapi::Schema::from(
-                openapi::ObjectBuilder::new().additional_properties(Some(openapi::Array::new(
-                    openapi::Ref::from_schema_name("OntologyOutwardEdges"),
-                ))),
-            )))
-            .into()
-    }
+#[derive(Default, Debug)]
+pub struct Edges {
+    ontology: HashMap<OntologyTypeEditionId, HashSet<OntologyOutwardEdges>>,
+    knowledge_graph: HashMap<EntityEditionId, HashSet<KnowledgeGraphOutwardEdges>>,
 }
 
-#[derive(Default, Debug, Serialize)]
-#[serde(transparent)]
-pub struct KnowledgeGraphRootedEdges(
-    // TODO: expose it through methods instead of making this field `pub`
-    //   see https://app.asana.com/0/1202805690238892/1203358665695491/f
-    pub HashMap<EntityId, HashMap<EntityVersion, HashSet<KnowledgeGraphOutwardEdges>>>,
-);
-
-// This is needed because Utoipa doesn't know how to handle the BTreeMaps
-impl ToSchema for KnowledgeGraphRootedEdges {
-    fn schema() -> openapi::Schema {
-        openapi::ObjectBuilder::new()
-            .additional_properties(Some(openapi::Schema::from(
-                openapi::ObjectBuilder::new().additional_properties(Some(openapi::Array::new(
-                    openapi::Ref::from_schema_name("KnowledgeGraphOutwardEdges"),
-                ))),
-            )))
-            .into()
+impl Edges {
+    #[must_use]
+    pub fn into_utoipa(self) -> crate::api::utoipa::subgraph::Edges {
```

Comment:
Can we call these `into_rest_repr` or something? This just doesn't make sense to me, utoipa defines type definitions.

Reply:
Moved to an `From` implementation 3ba34c780b8c39d05b9a08e78f767f5ac8dbb367